plugins/modules/swim_workflow_manager.py

```diff
@@ -752,30 +876,89 @@ def get_device_uuids(self, site_name, device_family, device_role, device_series_
+                    op_modifies=True,
+                    params=site_params,
+                )
 
-        self.log("Received API response from 'get_membership': {0}".format(str(response)), "DEBUG")
-        response = response['device']
+            except Exception as e:
+                self.log("Unable to fetch the device(s) associated to the site '{0}' due to '{1}'".format(site_name, str(e)), "WARNING")
+                return device_uuid_list
 
-        site_response_list = []
-        for item in response:
-            if item['response']:
-                for item_dict in item['response']:
-                    site_response_list.append(item_dict)
+            self.log("Received API response from 'get_membership': {0}".format(str(response)), "DEBUG")
+            response = response['device']
+
+            site_response_list = []
+            for item in response:
+                if item['response']:
+                    for item_dict in item['response']:
+                        site_response_list.append(item_dict)
+            self.log("family = getmembership version 2.3.5.3")
+        else:
+            self.msg = (
+                "'An exception occurred: In this version '{0}' 'get_membership' is not supported'".format(self.payload.get("dnac_version")))
+            self.log(self.msg, "ERROR")
+            self.module.fail_json(msg=self.msg)
+
+        if self.dnac_version >= self.version_2_3_7_6:
+            self.log("inside the version 2.3.7.6")
+            try:
+                response = self.dnac._exec(
+                    family="site_design",
+                    function='get_site_assigned_network_devices',
+                    op_modifies=True,
+                    params={"site_id": site_id},
+                )
+                self.log("get_site_assigned_network_device")
+                self.log(response)
+
+            except Exception as e:
+                self.log("Unable to fetch the device(s) associated to the site '{0}' due to '{1}'".format(site_name, str(e)), "WARNING")
+                return device_uuid_list
+
+            self.log("Received API response from 'get_site_assigned_network_devices': {0}".format(str(response)), "DEBUG")
+            response = response['response']
+
+            device_id_list = []
+            site_response_list = []
+
+            for device_id in response:
+                device_id_list.append(device_id.get("deviceId"))
+
+            for device_id in device_id_list:
+                param = {"id": device_id, "family": device_family}
+                try:
+                    response = self.dnac._exec(
+                        family="devices",
+                        function='get_device_list',
+                        op_modifies=True,
+                        params=param,
+                    )
+                    self.log("get device list response")
+                    self.log(response)
+                    site_response_list.append(response.get("response")[0])
```

Comment:
name is confusing as you are storing the device details in site_response_list.

Reply:
We attempted to replicate the get_membership API we have used 2 APIs , and to minimize code duplication, we used the same list variable name for the subsequent process.